UI/src/view/LinearAssetLabel.js

```diff
@@ -135,4 +135,91 @@
         };
 
     };
+
+    root.MassLimitationsLabel = function () {
+
+      AssetLabel.call(this);
+      var me = this;
+
+      var backgroundStyle = function (value, counter) {
+        return new ol.style.Style({
+          image: new ol.style.Icon(({
+            src: getImage(value),
+            anchor : [0.5, 1 + counter]
+          }))
+        });
+      };
+
+      var textStyle = function (value) {
+        if (!me.isValidValue(value))
+          return '';
+        // conversion Kg -> t
+        return ''.concat(value/1000, 't');
+      };
+
+      this.isValidValue = function (value) {
+        return value;
+      };
+
+      this.getStyle = function (asset, counter) {
+        return [backgroundStyle(getTypeId(asset), counter), new ol.style.Style({
+          text: new ol.style.Text({
+            text: textStyle(me.getValue(asset)),
+            fill: new ol.style.Fill({
+              color: '#000000'
+            }),
+            font: '14px sans-serif',
+            offsetY: getTextOffset(asset, counter)
+          })
+        })];
+      };
+
+      var getImage = function (typeId) {
+        var images = {
+          30: 'images/mass-limitations/totalWeightLimit.png'   ,
+          40: 'images/mass-limitations/trailerTruckWeightLimit.png',
+          50: 'images/mass-limitations/axleWeightLimit.png',
+          60: 'images/mass-limitations/bogieWeightLimit.png'
+        };
+        return images[typeId];
+      };
+
+
+      var getTextOffset = function (asset, counter) {
+        var offsets = { 30: -17 - (counter * 35), 40: -12 - (counter * 35), 50: -20 - (counter * 35), 60: -20 - (counter * 35)};
+        return offsets[getTypeId(asset)];
+      };
+
+      var getValues = function (asset) {
+        return asset.values;
+      };
+
+      this.getValue = function (asset) {
+        return asset.value;
+      };
+
+      var getTypeId = function (asset) {
+        return asset.typeId;
+      };
+
+      this.renderFeatures = function (assets, zoomLevel, getPoint) {
+        if (!me.isVisibleZoom(zoomLevel))
+          return [];
+
+
+        return [].concat.apply([], _.chain(assets).map(function (asset) {
+          var values = getValues(asset);
+          var size = values.length;
```

Comment:
Instead of using this variable, this could be done like: ` _.map(values, function (assetValues, **index**) {`